### PR TITLE
Autoprefix to PostCSS conversion

### LINKED ISSUE #8339: Update Autoprefixer browser versions
https://github.com/wet-boew/wet-boew/blob/5fe83b74fcd56e92a494c41426dc294638d96623/Gruntfile.coffee#L669-L676
Currently outputting CSS for very old browsers, so we should probably sync this with the currently supported browsers

### PR DESCRIPTION
**Pull request by nschonni**
# I included the built CSS temporarily just to show the changes at each stage with `WIP` comments. They need to be rebased out before landing

- Turns out `grunt-autoprefixer` has been replaced with `grunt-postcss` so I converted the plug-in
- I needed to update `grunt-sass` because the current version doesn't support Node 9. This upgrades to node-sass 4.8.3 with the LibSass 3.5 release rather than Beta 2
- Dropped old IE under 11
- Dropped old Android
- The recommend using a new `.browserlistrc` https://github.com/postcss/autoprefixer#browsers but I left that out

Closes #8339

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,3 +37,6 @@ csslint.out.xml
 *.orig
 *.tmp
 *.log
+
+!dist/theme-wet-boew/css/theme.css
```
Comment (nschonni) [inline .gitignore]: Rebase out

File: dist/theme-wet-boew/css/ie8-theme.css [hunk 2]
```diff
@@ -0,0 +1,9960 @@
+@charset "utf-8";
```
Comment (nschonni) [inline dist/theme-wet-boew/css/ie8-theme.css]: Rebase out

File: package.json [hunk 3]
```diff
@@ -3,8 +3,8 @@
   "version": "4.0.28-development",
   "description": "Web Experience Toolkit (WET): Open source code library for building innovative websites that are accessible, usable, interoperable, mobile-friendly and multilingual. This collaborative open source project is led by the Government of Canada. / Boîte à outils de l’expérience Web (BOEW) : Une bibliothèque de code primée pour construire des sites Web innovants qui sont accessibles, faciles d'emploi, interopérables, optimisés pour les appareils mobiles et multilingues. Ceci est un projet à source ouverte collaboratif dirigé par le Gouvernement du Canada.",
   "main": "index.html",
-  "engines" : {
-    "npm" : "~4.0.0"
+  "engines": {
```
Comment (nschonni) [inline package.json]: This was just bad whitespace from before

File: dist/theme-wet-boew/css/theme.css [hunk 4]
```diff
@@ -0,0 +1,10144 @@
+@charset "utf-8";
```
Comment (nschonni) [inline dist/theme-wet-boew/css/theme.css]: Rebase out

File: package-lock.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "wet-boew",
-  "version": "4.0.27-development",
+  "version": "4.0.28-development",
```
Comment (nschonni) [inline package-lock.json]: Aside: This probably should get updated next time as part of the release process

**Review by nschonni — COMMENTED** [review]
Adding comments for the parts to remove before landing and some changes I came across doing this

File: Gruntfile.coffee [hunk 6]
```diff
@@ -700,14 +701,15 @@ module.exports = (grunt) ->
...
-						"ie >= 8"
-						"ios 5"
-						"opera 12.1"
+					processors: [
+						require("autoprefixer")(
+							browsers: [
+								"last 2 versions"
+								"bb >= 10"
+								"Firefox ESR"
+								"ie > 10"
```
Comment (nschonni) [inline Gruntfile.coffee]: I can add back the IE8 here if you want, since this bundle includes the polyfills that are used for old IE

**Review by LaurentGoderre — APPROVED** [review]
LGTM

File: Gruntfile.coffee (hunk 6 above)
Comment (duboisp) [reply Gruntfile.coffee]: No need for that, we don't support IE8.

**Review by duboisp — APPROVED** [review]
I tested in IE11 on window 7 and seems all good.

**Review by EricDunsworth — APPROVED** [review]
I built it locally and browsed through random pages in Firefox Dev Edition/IE11/Opera. Things looked fine from what I could tell. Didn't do any head-to-head comparisons of generated CSS files though.

**Comment by EricDunsworth** [comment]
Btw once this is merged in, there might be value in updating the [roadmap page](http://wet-boew.github.io/v4.0-ci/docs/versions/rdmp-en.html) to clarify that WET no longer supports IE10. There might also be value in creating and maintaining a dedicated page in the documentation that gives an overview of all of WET's supported browsers.

**Comment by BenTrem** [comment]
FWIW _viz_.: https://en.wikipedia.org/wiki/Usage_share_of_web_browsers

**Pull request merged**